psi4/src/psi4/libpsipcm/psipcm.cc

```diff
@@ -157,31 +155,21 @@ PCM::PCM(int print_level, std::shared_ptr<BasisSet> basisset) {
     tess_charges_n_ = new double[ntess_];
     tess_charges_ = new double[ntess_];
 
-    auto atom_Zxyz_ = std::make_shared<Matrix>("Atom Zxyz", natom, 4);
-    for (int atom = 0; atom < natom; ++atom) {
-        Vector3 xyz = molecule->xyz(atom);
-        atom_Zxyz_->set(atom, 0, molecule->charge(atom));
-        atom_Zxyz_->set(atom, 1, xyz[0]);
-        atom_Zxyz_->set(atom, 2, xyz[1]);
-        atom_Zxyz_->set(atom, 3, xyz[2]);
-    }
-
     // The charge and {x,y,z} coordinates (in bohr) for each tessera
     tess_Zxyz_ = std::make_shared<Matrix>("Tess Zxyz", ntess_, 4);
     double **ptess_Zxyz = tess_Zxyz_->pointer();
     // Set the tesserae's coordinates (note the loop bounds; this function is 1-based)
     for (int tess = 1; tess <= ntess_; ++tess) pcmsolver_get_center(context_.get(), tess, &(ptess_Zxyz[tess - 1][1]));
 
     // Compute the nuclear potentials at the tesserae
-    double **patom_Zxyz = atom_Zxyz_->pointer();
-    ::memset(tess_pot_n_, 0, ntess_ * sizeof(double));
+    std::fill(tess_pot_n_, tess_pot_n_ + ntess_, 0.0);
```

Comment:
Or better yet, use `SharedVector` or `std::vector` over raw pointers.

Reply:
It's not really a problem (I think) since we're adhering to RAII here and the lifetime of these resources is tied to constructor and destructor. I would like to transition to `Vector`, however, because it can keep track of the symmetry blocking. I need to figure out how to get the symmetry information in and out of PCMSolver first, though. I have it working for DALTON, but the symmetry handling is slightly different between the two codes.

Reply:
It is not really a problem, but your `memset` was perfectly valid as well. I just noted that you modernized one part of the code, but left the underlying data structure unchanged when it would probably be about the same amount of work to just update the underlying data model.